barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/mega_flavor.hpp

```diff
@@ -467,6 +467,8 @@ class MegaFlavor {
             this->log_circuit_size = numeric::get_msb(this->circuit_size);
             this->num_public_inputs = proving_key.num_public_inputs;
             this->pub_inputs_offset = proving_key.pub_inputs_offset;
+            this->contains_recursive_proof = proving_key.contains_recursive_proof;
```

Comment:
actual core changes of PR, which is adding these two fields to the verification keys